src/backend/distributed/worker/worker_drop_protocol.c

```diff
@@ -34,6 +34,7 @@
 #include "utils/fmgroids.h"
 
 PG_FUNCTION_INFO_V1(worker_drop_distributed_table);
+PG_FUNCTION_INFO_V1(worker_drop_shell_table);
```

Comment:
Self-note / think more: Drop trigger calls `worker_drop_distributed_table`. 

So, for example if a table drop cascades into others, we still call that. For extension owned tables, dropping extension seems to work fine because of this.

For other, lets think more (e.g., drop parent)